Src/FluentAssertions/Collections/CollectionAssertions.cs

```diff
@@ -162,14 +162,25 @@ public AndConstraint<TAssertions> NotContainNulls(string because = "", params ob
                     .FailWith("Expected {context:collection} not to contain nulls{reason}, but collection is <null>.");
             }
 
-            object[] values = Subject.Cast<object>().ToArray();
-            for (int index = 0; index < values.Length; index++)
+            IEnumerable<int> indices = Subject
+                .Cast<object>()
+                .Select((item, index) => new { Item = item, Index = index })
+                .Where(e => ReferenceEquals(e.Item, null))
+                .Select(e => e.Index);
+
+            if (indices.Any())
             {
-                if (ReferenceEquals(values[index], null))
+                if (indices.Count() > 1)
```

Comment:
There are 15 places where we do the check `Count() > 1` only to know it a sequence has multiple items. (About 8 of them is just using `Count()` on an array, so they will use the `Length` property)

Would you prefer materializing the `IEnumerable` in an array or make an extension method like `HasMultipleItem()`?
That extension method could either be as simple as `list.Skip(1).Any()` or more sophisticated like
https://github.com/morelinq/MoreLINQ/blob/master/MoreLinq/CountMethods.cs